src/Features/Core/Portable/ExtractMethod/MethodExtractor.VariableSymbol.cs

```diff
@@ -21,294 +17,172 @@ internal abstract partial class AbstractExtractMethodService<
-                }
-
-                return left.DisplayOrder - right.DisplayOrder;
+                    (true, false) => 1,
+                    (false, true) => -1,
+                    // Then order by the general class of the variable (parameter, local, range-var).
+                    _ when (this._displayOrder != other._displayOrder) => this._displayOrder - other._displayOrder,
+                    // Finally, compare within the general class of the variable.
+                    _ => this.CompareToWorker(other),
+                };
             }
-        }
-
-        protected abstract class NotMovableVariableSymbol(ITypeSymbol type) : VariableSymbol(type)
-        {
-            public override SyntaxToken GetOriginalIdentifierToken(CancellationToken cancellationToken)
-                => default;
 
-            public override SyntaxAnnotation IdentifierTokenAnnotation => throw ExceptionUtilities.Unreachable();
-
-            public override void AddIdentifierTokenAnnotationPair(
-                MultiDictionary<SyntaxToken, SyntaxAnnotation> annotations, CancellationToken cancellationToken)
-            {
-                // do nothing for parameter
-            }
+            public string Name => this.GetSymbol().ToDisplayString(
+                new SymbolDisplayFormat(
+                    parameterOptions: SymbolDisplayParameterOptions.IncludeName,
+                    miscellaneousOptions: SymbolDisplayMiscellaneousOptions.EscapeKeywordIdentifiers));
         }
 
-        protected sealed class ParameterVariableSymbol : NotMovableVariableSymbol, IComparable<ParameterVariableSymbol>
+        protected abstract class VariableSymbol<TVariableSymbol, TSymbol>(
+            TSymbol symbol, ITypeSymbol symbolType, int displayOrder)
+            : VariableSymbol(symbolType, displayOrder)
+            where TVariableSymbol : VariableSymbol<TVariableSymbol, TSymbol>
+            where TSymbol : ISymbol
         {
-            private readonly IParameterSymbol _parameterSymbol;
+            protected TSymbol Symbol { get; } = symbol;
 
-            public ParameterVariableSymbol(IParameterSymbol parameterSymbol, ITypeSymbol type)
-                : base(type)
-            {
-                Contract.ThrowIfNull(parameterSymbol);
-                _parameterSymbol = parameterSymbol;
-            }
+            protected override ISymbol GetSymbol() => this.Symbol;
 
-            public override int DisplayOrder => 0;
+            protected sealed override int CompareToWorker(VariableSymbol right)
+                => this.CompareTo((TVariableSymbol)right);
 
-            protected override int CompareTo(VariableSymbol right)
-                => CompareTo((ParameterVariableSymbol)right);
+            protected abstract int CompareTo(TVariableSymbol right);
 
-            public int CompareTo(ParameterVariableSymbol other)
+            protected static int DefaultCompareTo(ISymbol left, ISymbol right)
```

Comment:
>DefaultCompareTo [](http://example.com/codeflow?start=33&length=16)

Why not have CompareTo be virtual and have this implementation (with ParameterVariableSymbol.CompareTo calling into the base instead of DefaultCompareTo)?

Reply:
i prefered forcing the subclasses to have to override, and decide what behavior they want :)